### PR TITLE
Ingest traces using COPY FROM

### PR DESCRIPTION
**Pull request by niksajakovljevic**
Using Postgres COPY FROM resulted in 2X improvements for trace ingest performance.
Here we use COPY FROM for inserting spans, links and events into respective hypertables.
COPY FROM uses binary protocol so it's expected to perform well.

### CODE REVIEW

**Review by jgpruitt — APPROVED** [review]
Why was exposing Acquire necessary? Why not use the CopyFrom method on the PgxConn interface?

**Comment by jgpruitt** [comment]
@niksajakovljevic do you plan to do the same optimization for ingesting the tags?

**Comment by niksajakovljevic** [comment]
>Why was exposing Acquire necessary? Why not use the CopyFrom method on the PgxConn interface?

Mainly because I wanted to explicitly control the transaction.

>do you plan to do the same optimization for ingesting the tags?

I left that out for this PR mainly because I focused on part that deals with most data. However if we discover that the amount of tags is high we can do the same there...

**Review by antekresic — APPROVED** [review]
Some minor stuff but mostly looks good to me! :+1:

File: pkg/pgmodel/ingestor/trace/writer.go [hunk 1]
```diff
@@ -37,15 +39,11 @@ const (
...
-	insertSpanLinkSQL = `INSERT INTO _ps_trace.link (trace_id, span_id, span_start_time, linked_trace_id, linked_span_id, trace_state, tags, dropped_tags_count, link_nbr)
-		VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9)`
-	insertSpanEventSQL = `INSERT INTO _ps_trace.event (time, trace_id, span_id, name, event_nbr, tags, dropped_tags_count)
-		VALUES ($1, $2, $3, $4, $5, $6, $7)`
-	insertSpanSQL = `INSERT INTO _ps_trace.span (trace_id, span_id, trace_state, parent_span_id, operation_id, start_time, end_time, span_tags, dropped_tags_count,
-		event_time, dropped_events_count, dropped_link_count, status_code, status_message, instrumentation_lib_id, resource_tags, resource_dropped_tags_count, resource_schema_url_id)
-		VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13, $14, $15, $16, $17, $18)
-		ON CONFLICT DO NOTHING`  // Most cases conflict only happens on retries, safe to ignore duplicate data.
+var (
+	linkColumns  = []string{"trace_id", "span_id", "span_start_time", "linked_trace_id", "linked_span_id", "trace_state", "tags", "dropped_tags_count", "link_nbr"}
```
[the remark this answers is not in the document]
Comment (niksajakovljevic) [reply pkg/pgmodel/ingestor/trace/writer.go:43]: Maybe `linkTableColumns`?

**Pull request merged**